source/main.py

```diff
@@ -0,0 +1,307 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import os
```

Comment:
most of these imports should be in another function, unless you absolutely need them in here